sdjournal/read.go

```diff
@@ -103,33 +106,42 @@ func NewJournalReader(config JournalReaderConfig) (*JournalReader, error) {
 
 func (r *JournalReader) Read(b []byte) (int, error) {
 	var err error
-	var c int
 
-	// Advance the journal cursor
-	c, err = r.journal.Next()
+	if r.msgReader == nil {
+		var c int
 
-	// An unexpected error
-	if err != nil {
-		return 0, err
-	}
+		// Advance the journal cursor. It has to be called at least one time
+		// before reading
+		c, err = r.journal.Next()
 
-	// EOF detection
-	if c == 0 {
-		return 0, io.EOF
-	}
+		// An unexpected error
+		if err != nil {
+			return 0, err
+		}
 
-	// Build a message
-	var msg string
-	msg, err = r.buildMessage()
+		// EOF detection
+		if c == 0 {
+			return 0, io.EOF
+		}
 
-	if err != nil {
-		return 0, err
+		// Build a message
+		var msg string
+		msg, err = r.buildMessage()
+
+		if err != nil {
+			return 0, err
+		}
+		r.msgReader = strings.NewReader(msg)
 	}
 
 	// Copy and return the message
-	copy(b, []byte(msg))
+	var sz int
+	sz, err = r.msgReader.Read(b)
+	if sz == r.msgReader.Len() {
+		r.msgReader = nil
```

Comment:
@alban this seems to generate some gc pressure. Not sure if that's going to be a big deal.
Wonder if we can update the string instead of creating the new reader struct every time we read?